src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,327 @@
+     */
+    private static final int PATH_CLOSE                = 3;
+
+    /**
+     * The distance the guard is searching entities in currently.
+     */
+    private int currentSearchDistance = START_SEARCH_DISTANCE;
+
+    /**
+     * The current target.
+     */
+    protected EntityLivingBase targetEntity;
+
+    /**
+     * Current goTo task.
+     */
+    private BlockPos currentPatrolTarget;
+
+    /**
+     * Containing all close entities.
+     */
+    private List<Entity> entityList;
+
+    /**
+     * Amount of arrows already shot or sword hits dealt.
+     */
+    protected int attacksExecuted = 0;
+
+    /**
+     * Sets up some important skeleton stuff for every ai.
+     *
+     * @param job the job class
+     */
+    protected AbstractEntityAIGuard(@NotNull final JobGuard job)
+    {
+        super(job);
+        super.registerTargets(
+                new AITarget(IDLE, () -> START_WORKING),
+                new AITarget(START_WORKING, () -> GUARD_RESTOCK)
+        );
+    }
+
+
+    /**
+     * Can be overridden in implementations.
+     * <p>
+     * Here the AI can check if the armour have to be re rendered and do it.
+     */
+    @Override
+    protected void updateRenderMetaData()
+    {
+        updateArmor();
+    }
+
+    /**
+     * Goes back to the building and tries to take armour from it when he hasn't in his inventory.
+     * @return the next state to go to.
+     */
+    protected AIState goToBuilding()
+    {
+        if(!walkToBuilding())
+        {
+            final AbstractBuildingWorker workBuilding = getOwnBuilding();
+            if(workBuilding != null)
+            {
+                final TileEntityColonyBuilding chest = workBuilding.getTileEntity();
+
+                for (int i = 0; i < workBuilding.getTileEntity().getSizeInventory(); i++)
+                {
+                    final ItemStack stack = chest.getStackInSlot(i);
+
+                    if (stack == null)
+                    {
+                        continue;
+                    }
+
+                    if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == null)
+                    {
+                        final int emptySlot = worker.getInventoryCitizen().getFirstEmptySlot();
+
+                        if(emptySlot != -1)
+                        {
+                            worker.getInventoryCitizen().setInventorySlotContents(emptySlot, stack);
+                            chest.setInventorySlotContents(i, null);
+                        }
+                    }
+                    else if(!(stack.getItem() instanceof ItemBow || Utils.doesItemServeAsWeapon(stack)))
+                    {
+                        //todo dump everything which isn't a weapon or armor
+                    }
+                }
+            }
+            attacksExecuted = 0;
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        return AIState.GUARD_RESTOCK;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'AIState.GUARD_RESTOCK' due to existing static import 'com.minecolonies.entity.ai.util.AIState.GUARD_RESTOCK' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)